tests/unit/jobs/UserJobsTest.php

```diff
@@ -244,5 +245,31 @@ public function canUpdateUser()
             array_only($userFromDb->attributesToArray(), $testedAttributes)
         );
     }
+
+    /**
+     * @test
+     */
+    public function cannotCreateUserTwiceWithSameEmail()
+    {
+        $user = dispatch_now(new CreateUser([
+            'email'         => 'john.doe@example.com',
+            'password'      => 'secret',
+            'first_name'    => 'John',
+            'last_name'     => 'Doe',
+        ]));
+
+        try {
+            $user = dispatch_now(new CreateUser([
+                'email'         => 'john.doe@example.com',
+                'password'      => 'secret2',
+                'first_name'    => 'John2',
+                'last_name'     => 'Doe2',
+            ]));
+        } catch (QueryException $e) {
```

Comment:
I think that we should catch `Exception` here and assert that it's a `QueryException` instance.